Lib/inspect.py

```diff
@@ -913,11 +914,18 @@ def tokeneater(self, type, token, srowcol, erowcol, line):
             self.passline = True    # skip to the end of the line
         elif token == "(":
             if self.indecorator:
+                self.decoratorparens.append(token)
                 self.decoratorhasargs = True
         elif token == ")":
             if self.indecorator:
-                self.indecorator = False
-                self.decoratorhasargs = False
+                if self.decoratorparens:
+                    if self.decoratorparens[-1] == "(":
+                        self.decoratorparens = self.decoratorparens[:-1]
+                    if not self.decoratorparens:
+                        self.indecorator = False
+                        self.decoratorhasargs = False
+                else:
+                    self.decoratorparens.append(token)
```

Comment:
(Why) Is it necessary to keep track of the tokens?
(came here via https://github.com/python/cpython/pull/21425, which only uses a count)